public/app/features/dashboard/state/initDashboard.ts

```diff
@@ -206,6 +208,25 @@ export function initDashboard(args: InitDashboardArgs): ThunkResult<void> {
     }
 
     // yay we are done
+    if (dashDTO.dashboard.uid) {
+      dashboard.panels.forEach((panel) => {
+        panel.targets.forEach((target) => {
+          let dsSpecific = {};
+          if (target.datasource?.type === 'grafana-azure-monitor-datasource') {
+            dsSpecific = getAzureMonitorEvent(target as AzureMonitorQuery);
```

Comment:
nit: is there a way to be stricter here with the type checking rather than asserting the type?

Reply:
This is a tough one... 🤔 
I think the new logic (changing the datasource API to have a new endpoint that returns the event to emit) will fix this though as we won't have to cast the target here.